### PR TITLE
Do not make checksums of checksums

### LINKED ISSUE #3: make_check_sums create checksums of checksums
Each time I'm calling it, it creates a whole new generation of checksums.

### PR DESCRIPTION
**Pull request by Zeitsperre**
Fixes #3

### CODE REVIEW

File: make_check_sums.py [hunk 1]
```diff
@@ -6,13 +6,17 @@
...
-    files = [f for f in Path().cwd().rglob("*") if f.parent.name != "raven-testdata"]
+    files = [
+        f
+        for f in Path().cwd().rglob("*")
+        if f.parent.name != "raven-testdata" and not str(f).endswith(".md5")
+    ]
     for f in files:
-        outf = f'{f.as_posix()}.md5'
+        outf = f"{f.as_posix()}.md5"
         if f.is_file() and not Path(outf).exists():
```
Comment (huard) [inline make_check_sums.py:16]: I suggest you review this logic instead. Ie remplace the file suffix by md5, then check if it exists.

**Review by huard — CHANGES_REQUESTED** [review]

**Review by huard — COMMENTED** [review]
Every md5 will be regenerated. Won't git think that all the md5 files were changed ? 
In the loop, 
#. if suffix == ".md5" continue
#. if file + ".md5" does not exist create checksum

**Comment by Zeitsperre** [comment]
In the initial case, they were regenerated, but because there are no changes (the `.md5` string doesn't change), they're seen as the exact same file, so they're ignored by git.

I'm still a bit lost as I thought that the logic in my initial commit to skip the processing of files with existing `.md5` accompanying files was enough, ie:
```
...
if f.parent.name != "raven-testdata" and not str(f).endswith(".md5")
)

Am I missing something?
```

**Comment by huard** [comment]
1. I thought it was weird to use Path object, but then use string.endswith to check the suffix.
2. The list comprehension bundles a lot of stuff (glob pattern + two logical conditions). Also why the parent.name condition ? 

So basically, my comment is "make the logic stand-out", this way, we won't trip ourselves in it. This is a trivial example, but this is useful everywhere.

**Comment by Zeitsperre** [comment]
>     1. I thought it was weird to use Path object, but then use string.endswith to check the suffix.

That's reasonable. I think Path().suffix is probably fine as well. I can integrate that into this PR.

>     2. The list comprehension bundles a lot of stuff (glob pattern + two logical conditions)

I can pull the logic out of there if we want. It's a useful aspect of lost comprehensions, so I figured why not use it. The alternative I have in mind is to find all objects that end with md5 and all those that don't (and are not folders), sanitize the `".md5"` from the `.md5` files, turn them into sets and then make `.md5` files on the non-`.md5` files that differ between them. Would that be closer to what you were envisioning ?

> Also why the parent.name condition ?

If that isn't there, we end up making pointless `.md5` files for the top-level files (e.g. `make_check_sums.py.md5`).

**Comment by huard** [comment]
1. OK
2. I love list comprehensions, but they quickly get unclear if you include too many logical conditions. Yes. 

ok

**Comment by Zeitsperre** [comment]
Good to go?

**Comment by huard** [comment]
The code created checksums for hidden files (.e.g. .git).

**Review by huard — APPROVED** [review]

**Comment by Zeitsperre** [comment]
Wow, I didn't even realize. My bad!

**Pull request merged**